src/math/data2d.cpp

```diff
@@ -355,27 +356,23 @@ bool Data2D::deserialise(LineParser &parser)
     // Read errors / valuse
     if (hasError_)
     {
-        for (auto x = 0; x < x_.size(); ++x)
+        ArrayIndex2D index(values_);
+        while (index++.hasNext())
         {
-            for (auto y = 0; y < y_.size(); ++y)
-            {
-                if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
-                    return false;
-                values_[{x, y}] = parser.argd(0);
-                errors_[{x, y}] = parser.argd(1);
-            }
+            if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+                return false;
+            values_[*index] = parser.argd(0);
+            errors_[*index] = parser.argd(1);
```

Comment:
With my changes suggested above, we get a little cleaner code here
```suggestion
            values_[index] = parser.argd(0);
            errors_[index] = parser.argd(1);
```